src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2015 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import com.intellij.plugins.haxe.HaxeComponentType;
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.plugins.haxe.model.type.HaxeTypeResolver;
+import com.intellij.plugins.haxe.model.type.SpecificHaxeClassReference;
+import com.intellij.plugins.haxe.model.type.SpecificTypeReference;
+import com.intellij.plugins.haxe.util.HaxePsiUtils;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiIdentifier;
+import org.apache.commons.lang.NotImplementedException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+public class HaxeClassModel {
+  public HaxeClass haxeClass;
+
+  public HaxeClassModel(HaxeClass haxeClass) {
+    this.haxeClass = haxeClass;
+  }
+
+  public HaxeClassReferenceModel getParentClassReference() {
+    List<HaxeType> list = haxeClass.getHaxeExtendsList();
+    if (list.size() == 0) return null;
+    return new HaxeClassReferenceModel(list.get(0));
+  }
+
+  static public boolean isValidClassName(String name) {
+    return name.substring(0, 1).equals(name.substring(0, 1).toUpperCase());
```

Comment:
Just because the first letter is capitalized doesn't mean that the class name is valid.  That's only one requirement.  For example, it can't start with a number or an underscore, and it can only contain letters, numbers, and underscores.